src/core/draw_tools/components/DrawToolsToolbox/DrawToolsToolbox.tsx

```diff
@@ -38,29 +38,45 @@ export const DrawToolsToolbox = () => {
   }, [settings]);
 
   const onToggleDrawMode = useCallback(
-    (id) => toggleDrawMode(id),
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [],
+    (id) => {
+      toggleDrawMode(id);
+    },
+    [toggleDrawMode],
   );
 
   const buttons = useMemo(
     () => [
       settings.availableModes?.includes('DrawPolygonMode') && (
-        <Button id={drawModes.DrawPolygonMode} dark variant="invert">
+        <Button
+          id={drawModes.DrawPolygonMode}
+          dark
+          variant="invert"
+          onClick={() => onToggleDrawMode(drawModes.DrawPolygonMode)}
```

Comment:
you can just use toggleDrawMode as it will be new callback every render 

`const onToggleDrawMode = useCallback(`
this one can be removed at all